plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -0,0 +1,577 @@
+    -- governance scripts to be able to see the values in the PParamsUpdate, but I am not
+    -- sure how to go forward about this, so for now I leave it as empty.
+    ParameterChange -- Should contain: (PParamsUpdate era)
+    --
+  | -- | proposal to update protocol version
+    HardForkInitiation ProtocolVersion
+  | TreasuryWithdrawals !(Map V2.Credential V2.Value)
+  | NoConfidence
+  | NewCommittee
+      -- | Old committee
+      [ColdCommitteeHash]
+      -- | New Committee
+      Committee
+  | NewConstitution Constitution
+  | InfoAction
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow GovernanceAction)
+
+instance PlutusTx.Eq GovernanceAction where
+  {-# INLINEABLE (==) #-}
+  ParameterChange == ParameterChange = Haskell.True
+  HardForkInitiation a == HardForkInitiation a' = a PlutusTx.== a'
+  TreasuryWithdrawals a == TreasuryWithdrawals a' = a PlutusTx.== a'
+  NoConfidence == NoConfidence = Haskell.True
+  NewCommittee a b == NewCommittee a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+  NewConstitution a == NewConstitution a' = a PlutusTx.== a'
+  InfoAction == InfoAction = Haskell.True
+  _ == _ = Haskell.False
+
+data ProposalProcedure = ProposalProcedure
+  { ppDeposit          :: V2.Value
+  , ppReturnAddr       :: V2.Credential
+  , ppGovernanceAction :: GovernanceAction
+  , ppAnchor           :: Anchor
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty ProposalProcedure where
+  pretty ProposalProcedure{..} =
+    vsep
+      [ "ppDeposit:" <+> pretty ppDeposit
+      , "ppReturnAddr:" <+> pretty ppReturnAddr
+      , "ppGovernanceAction:" <+> pretty ppGovernanceAction
+      , "ppAnchor:" <+> pretty ppAnchor
+      ]
+
+instance PlutusTx.Eq ProposalProcedure where
+  {-# INLINEABLE (==) #-}
+  ProposalProcedure a b c d == ProposalProcedure a' b' c' d' =
+    a PlutusTx.== a'
+      PlutusTx.&& b PlutusTx.== b'
+      PlutusTx.&& c PlutusTx.== c'
+      PlutusTx.&& d PlutusTx.== d'
+
+data ScriptPurpose
+  = Minting V2.CurrencySymbol
+  | Spending V2.TxOutRef
+  | Rewarding V2.StakingCredential
+  | Certifying TxCert
+  | Voting Voter GovernanceActionId
+  | Proposing -- For now this will only be used by Constitution, thus might turned out to be
+  -- unused.
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow ScriptPurpose)
+
+instance PlutusTx.Eq ScriptPurpose where
+  {-# INLINEABLE (==) #-}
+  Minting a == Minting a' =
+    a PlutusTx.== a'
+  Spending a == Spending a' =
+    a PlutusTx.== a'
+  Rewarding a == Rewarding a' =
+    a PlutusTx.== a'
+  Certifying a == Certifying a' =
+    a PlutusTx.== a'
+  Voting a b == Voting a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+  Proposing == Proposing = Haskell.True
+  _ == _ = Haskell.False
+
+-- | TxInfo for PlutusV3
+data TxInfo = TxInfo
+  { txInfoInputs                :: [V2.TxInInfo]
+  -- ^ Unchanged
```

Comment:
haddock should be informative about the current state, not the diffs!

Reply:
I don't know. Like I said above, I think these are useful for someone migrating from V2 to V3, which is a lot of people.

Reply:
I agree that this it is useful to have in haddock what has changed since previous version (v2 or v1)